app/src/main/java/com/ranveeraggarwal/letrack/activities/AddPersonActivity.java

```diff
@@ -37,24 +48,48 @@
     String selectedFrequency;
     String selectedStartDate;
 
+    int REQUEST_CONTACTS = 1;
+    String TAG = "AddPersonActivity";
+    boolean hasContactPermission = false;
+
     DatabaseAdapter databaseAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         setContentView(R.layout.activity_add_person);
+        if (ContextCompat.checkSelfPermission(
+                this, Manifest.permission.READ_CONTACTS) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.READ_CONTACTS}, REQUEST_CONTACTS);
+        } else {
+            hasContactPermission = true;
+        }
```

Comment:
What if I do want to use it later? We need to think of the case when a user clicks on deny by mistake. This feature will be hidden from then forever, in that case. Plus, moving it to `onClick` is relatively less intrusive since the permissions are asked only when the feature is about to be used. What do you think?